prow/github/client_test.go

```diff
@@ -38,6 +40,29 @@ func getClient(url string) *Client {
 	}
 }
 
+func TestRequestRateLimit(t *testing.T) {
+	var slept time.Duration
+	timeSleep = func(d time.Duration) { slept = d }
+	defer func() { timeSleep = time.Sleep }()
+	ts := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if slept == 0 {
+			w.Header().Set("X-RateLimit-Remaining", "0")
+			w.Header().Set("X-RateLimit-Reset", strconv.Itoa(int(time.Now().Add(time.Second).Unix())))
+			http.Error(w, "403 Forbidden", http.StatusForbidden)
+		}
+	}))
+	defer ts.Close()
+	c := getClient(ts.URL)
+	resp, err := c.request(http.MethodGet, c.base, nil)
+	if err != nil {
+		t.Errorf("Error from request: %v", err)
+	} else if resp.StatusCode != 200 {
```

Comment:
Shouldn't the three conditions here be decoupled, instead of in an if..else if...?